src/Humanizer/StringHumanizeExtensions.cs

```diff
@@ -29,12 +29,16 @@ private static string FromPascalCase(string input)
         {
             var result = string.Join(" ", PascalCaseWordPartsRegex
                 .Matches(input).Cast<Match>()
-                .Select(match => match.Value.ToCharArray().All(char.IsUpper) &&
-                    (match.Value.Length > 1 || (match.Index > 0 && input[match.Index - 1] == ' ') || match.Value == "I")
-                    ? match.Value
-                    : match.Value.ToLower()));
+                .Select(match =>
+                {
+                    var value = match.Value;
+                    return value.All(char.IsUpper) &&
+                           (value.Length > 1 || (match.Index > 0 && input[match.Index - 1] == ' ') || value == "I")
+                        ? value
+                        : value.ToLower();
+                }));
 
-            if (result.Replace(" ", "").ToCharArray().All(c => char.IsUpper(c)) &&
+            if (result.Replace(" ", "").All(char.IsUpper) &&
```

Comment:
Depending on compiler the method group can be slower than a lambda due to boxing to a delegate.

Reply:
does that effect us when we always compile our dll with the latests c# and sdk?

Reply:
I don’t know. Probably not. It’s just a habit from old days.